bears/python/PythonPackageInitBear.py

```diff
@@ -0,0 +1,29 @@
+import os
+
+from coalib.results.Result import Result
+from coalib.bears.GlobalBear import GlobalBear
+
+
+class PythonPackageInitBear(GlobalBear):
+    """
+    Looks for missing __init__.py files in directories containing python files.
+    """
+
+    LANGUAGES = {'Python', 'Python 3', 'Python 2'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    def run(self):
+        files = sorted(self.file_dict)
+        dirs = set()
+
+        for file in files:
+            dir = os.path.split(file)[0]
```

Comment:
dir  is a builtin, choose another name. Also this can be sped up performance wise and you can get rid of the need for `files`. Also we should potentially ignore non python files because there's no usecase where this would make sense and it's super easy:

```python
dirs = {os.path.split(file)[0] for file in self.file_dict.keys()
        if file.endswith('.py')}
missing_inits = {os.path.join(py_dir, '__init__.py')
                 for py_dir in dirs
                 if not os.path.join(py_dir, '__init__.py') in self.file_dict}

for init in missing_inits:
    yield Result(...)
```